string id
int64 lineNum
int64 state
int64 stateTime
string message
bool complete
